### DISCUSSION TITLE
If one of the replica is down, how long operation log can keep in ZooKeeper?

### DISCUSSION THREAD

**Discussion by wxl356** [ClickHouse/ClickHouse#28513] [Q&A] [answered]
Under my understanding, in ReplicatedMergeTree, insert operation will write log in ZK "/log", other replica pull log, execute task and sync date.

My question is if one replica is unavailable(machine is down or clickhouse instance is down), this replica cannot pull log and sync data. How long the operation log will keep in ZK? Is there valid period?  I want to know what clickhouse will do if one replica is down for a very long time.

Appreciate for any answer or discussion, thank you.

**Comment by tavplubix** [comment, marked as answer]
> How long the operation log will keep in ZK?

It's controlled by the following settings:

```
:) select * from system.merge_tree_settings where name in ('min_replicated_logs_to_keep', 'max_replicated_logs_to_keep') \G

Row 1:
──────
name:        max_replicated_logs_to_keep
value:       1000
changed:     0
description: How many records may be in log, if there is inactive replica. Inactive replica becomes lost when when this number exceed.
type:        UInt64

Row 2:
──────
name:        min_replicated_logs_to_keep
value:       10
changed:     0
description: Keep about this number of last records in ZooKeeper log, even if they are obsolete. It doesn't affect work of tables: used only to diagnose ZooKeeper log before cleaning.
type:        UInt64

```

> what clickhouse will do if one replica is down for a very long time.

Active replicas will mark staled replica as lost. When lost replica becomes available it clones state (parts set, replication queue, replication log pointer, columns metadata) of some alive replica.
